GeneralsMD/Code/CMakeLists.txt

```diff
@@ -29,8 +29,19 @@ target_include_directories(cnc-zero-hour-common PRIVATE
     ${CMAKE_CURRENT_SOURCE_DIR}/Libraries/Source/WWVegas/WWDebug
     # Required by DataChunk.cpp
     ${CMAKE_CURRENT_SOURCE_DIR}/Libraries/Source/Compression
-    ${CMAKE_CURRENT_SOURCE_DIR}/extern/dxvk-native/usr/include/dxvk
-    ${CMAKE_CURRENT_SOURCE_DIR}/extern/compat
 )
 
-target_compile_options(cnc-zero-hour-common PRIVATE -Wno-invalid-offsetof)
\ No newline at end of file
+if(WIN32)
+    add_subdirectory(extern/d3d8)
+    target_link_libraries(cnc-zero-hour-common PRIVATE d3d8)
+else()
+    target_include_directories(cnc-zero-hour-common PRIVATE
+        ${CMAKE_CURRENT_SOURCE_DIR}/extern/dxvk-native/usr/include/dxvk
+        ${CMAKE_CURRENT_SOURCE_DIR}/extern/compat
+    )
+endif()
+
+if(CMAKE_GENERATOR MATCHES "Visual Studio")    
```

Comment:
I'd prefer if the compiler was detected instead of the generator.
Otherwise this looks fine :+1: :)